pennylane/pulse/rydberg_machine.py

```diff
@@ -0,0 +1,237 @@
+        """
+        if self._rydberg_interaction is None:
+
+            def rydberg_projector(wire: int) -> SProd:
+                """Returns the projector into the Rydberg state for the given wire.
+
+                Args:
+                    wire (int): _description_
+
+                Returns:
+                    SProd: projector into the Rydberg state
+                """
+                return (1 - qml.PauliZ(wire)) / 2
+
+            coeffs = []
+            ops = []
+            for idx, (pos1, wire1) in enumerate(zip(self.coordinates[1:], self.wires[1:])):
+                for pos2, wire2 in zip(self.coordinates[: idx + 1], self.wires[: idx + 1]):
+                    atom_distance = np.linalg.norm(qml.math.array(pos2) - pos1)
+                    Vij = self.interaction_coeff / (
+                        abs(atom_distance) ** 6
+                    )  # van der Waals potential
+                    coeffs.append(Vij)
+                    ops.append(qml.prod(rydberg_projector(wire1), rydberg_projector(wire2)))
+
+            self._rydberg_interaction = qml.dot(coeffs, ops)
+
+        return self._rydberg_interaction
+
+    def local_drive(self, rabi: list, detunings: list, phases: list, wires: list):
+        """Apply ``N = len(rabi)``  local driving laser fields with the given rabi frequencies,
+        detunings and phases acting on the given wires.
+
+        Args:
+            rabi (list): list of Rabi frequency values (in MHz) of each driving laser field
+            detunings (list): list of detuning values (in MHz) of each driving laser field
+            phases (list): list of phases (in radiants) of each driving laser field
+            wires (list): list of wire values that each laser field acts on
+        """
+        lengths = [len(rabi), len(detunings), len(phases), len(wires)]
+        if len(set(lengths)) > 1:
+            raise ValueError(
+                "The lists containing the driving parameters must all have the same"
+                f"length. Got lengths: {lengths}"
+            )
+        if any(wire not in self.wires for wire in wires):
+            raise ValueError(
+                "The wires list contains a wire value that is not present in the RydbergMachine."
+            )
+        # Update `_driving_interaction` Hamiltonian
+        ops = [
+            qml.math.cos(p) * qml.PauliX(w) - qml.math.sin(p) * qml.PauliY(w)
+            for p, w in zip(phases, wires)
+        ]
+        H1 = qml.dot(rabi, ops)
+        H2 = qml.dot(detunings, [qml.PauliZ(w) for w in wires])
+        self._driving_interaction += 1 / 2 * H1 + 1 / 2 * H2
+        # Update dictionaries of applied pulses to allow translation into hardware
+        self._local_drives["rabi"].extend(rabi)
+        self._local_drives["detunings"].extend(detunings)
+        self._local_drives["phases"].extend(phases)
+        self._local_drives["wires"].extend(wires)
+
+    def global_drive(self, rabi: float, detuning: float, phase: float):
```

Comment:
These should have the option of being callables as well

Reply:
Updated! 👍